crates/accelerate/src/synthesis/permutation/utils.rs

```diff
@@ -144,3 +144,18 @@ pub fn decompose_cycles(cycles: &Vec<Vec<usize>>) -> Vec<(usize, usize)> {
 
     swaps
 }
+
+//
+pub fn create_swap_layer(pattern: &ArrayView1<usize>, starting_point: usize) -> Vec<(usize, usize)> {
+    let mut pattern_vec = pattern.to_vec();
+    let num_qubits = pattern_vec.len();
+    let mut gates = Vec::new();
+
+    for j in (starting_point..num_qubits).step_by(2) {
+        if pattern_vec[j] > pattern_vec[(j + 1) % num_qubits] {
+            gates.push((j, (j + 1) % num_qubits));
+            pattern_vec.swap(j, (j + 1) % num_qubits);
+        }
+    }
```

Comment:
You might want to use `&ArrayViewMut1<_>` here, the changes you're making to this pattern aren't being applied here.

Once you do so you shouldn't need to convert your `ArrayView` into a `vec` and the changes you're making to it should be applied in place. The way this is being done now does not save any of the swaps you're performing on `pattern`.